solarforecastarbiter/metrics/tests/test_preprocessing.py

```diff
@@ -1653,3 +1726,233 @@ def test_apply_fill_unstratified_dataframe(data, method, exp, exp_count,
                                              data.index[0], data.index[-1])
     pd.testing.assert_frame_equal(result, exp)
     assert count == exp_count
+
+
+OUTAGE_FORECAST = datamodel.Forecast(
+   site=_site_metadata(), name='dummy fx', variable='ghi',
+   interval_value_type='instantaneous',
+   interval_length=pd.Timedelta('1h'),
+   interval_label='beginning',
+   issue_time_of_day=dt.time(hour=5),
+   lead_time_to_start=pd.Timedelta('1h'),
+   run_length=pd.Timedelta('12h')
+)
+
+
+@pytest.mark.parametrize("forecast,start,end,expected", [
+    (OUTAGE_FORECAST,
+     pd.Timestamp('2021-01-01T00:00Z'),
+     pd.Timestamp('2021-01-02T00:00Z'),
+     pd.date_range(
+        '2020-12-31T05:00Z',
```

Comment:
This issue time doesn't contribute to the report. First one is `'2020-12-31T17:00Z'`, which has valid times `['2020-12-31T18:00Z', '2021-01-01T5:00Z')`

Reply:
corrected with updated algorithm